test/test_assemblers.py

```diff
@@ -40,6 +41,16 @@ def assertFilesystem(self, device, uuid, fstype, tree_id):
             self.assertEqual(diff["deleted_files"], [])
             self.assertEqual(diff["differences"], {})
 
+    def assertGRUB2(self, device, l1hash, l2hash, size):
+        m1 = hashlib.sha256()
+        m2 = hashlib.sha256()
+        with open(device, "rb") as d:
+            m1.update(d.read(440))
+            d.seek(512, 0)
```

Comment:
This is were the test fails. Works for me on Ubuntu, so that can't be it. `lseek(2)` says it returns `EINVAL` when the resulting file offset would be beyond the end of the seekable device. I can't imagine that happening for `512`, because the other tests passed.